### ISSUE TITLE
Http trigger serializes returned ValueTask<IActionResult> as POCO

### ISSUE THREAD

**Issue by andrewjw1995** [Azure/azure-functions-host#7120]
Http triggers allow you to return a `Task<IActionResult>`, and the host will await the task and execute the result. However, if I return a `ValueTask<IActionResult>` instead, the whole value task is serialized:

    [FunctionName("Foo")]
    public ValueTask<IActionResult> ExecuteAsync([HttpTrigger(AuthorizationLevel.Function, "get")]HttpRequest request)
    {
        return new ValueTask<IActionResult>(OkObjectResult("Foo"));
    }

Expected output:

    "Foo"

Actual output:

    {
        "isCompleted": false,
        "isCompletedSuccessfully": false,
        "isFaulted": false,
        "isCanceled": false,
        "result": {
            "value": "Foo",
            "formatters": [],
            "contentTypes": [],
            "declaredType": null,
            "statusCode": 200
        }
    }

In most cases the return type of the function can simply be changed from `ValueTask` to `Task`. However it is not the behaviour I expected, and since there is already a check for `Task`, I imagine it wouldn't be hard to also check for `ValueTask`. I'm happy to implement this if I'm allowed.

 - Microsoft.NET.Sdk.Functions 3.0.11

**Comment by v-anvari** [comment]
Hi @andrewjw1995 , Apologies for the delayed response, have you tried using OkResult. 

![image](https://user-images.githubusercontent.com/66122429/123454723-8550b300-d5fe-11eb-8d7a-82096ba913bf.png)


https://docs.microsoft.com/en-us/dotnet/api/system.web.http.apicontroller.ok?view=aspnetcore-2.1

For Example:
```
 [FunctionName("Function1")]
        public static async Task<IActionResult> Run(
            [HttpTrigger(AuthorizationLevel.Function, "get", "post", Route = "Function1/{id}")] FooQuery Foo,
            ILogger log)
        {
            log.LogInformation("C# HTTP trigger function processed a request.");

            return new OkResult();
        }

        public class FooQuery
        {
            public string Id { get; set; }
            public string Name { get; set; }
        }
```

You are calling OkObjectResult, which will return specified values with 200 response. I hope this has resolved your query.

**Comment by andrewjw1995** [comment]
@v-anvari No, I have not tried using `OkResult`, as that is the wrong class for my use case. From the documentation:

> [OkResult][1]: An StatusCodeResult that when executed will produce an empty Status200OK response.

> [OkObjectResult][2]: An ObjectResult that when executed performs content negotiation, formats the entity body, and will produce a Status200OK response if negotiation and formatting succeed.

Since I am trying to return content in the body of the response, `OkObjectResponse` is the correct class to use, _not_ `OkResult`.

[1]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.okresult?view=aspnetcore-2.1
[2]: https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.okobjectresult?view=aspnetcore-2.1

**Comment by v-anvari** [comment]
Hi @andrewjw1995 , Thank you for your feedback! We will investigate this further and update as appropriate.

**Comment by v-anvari** [comment]
Hi @andrewjw1995 , Apologies, checked this behavior internally. This is a known limitation in Functions. You will have to return Task instead of valueTask, the output will be serialized by using ValueTask. 

Adding this issue to enhancement to include this behavior in the future versions
